src/vcpkg.cpp

```diff
@@ -52,6 +57,44 @@ namespace
         Checks::exit_fail(VCPKG_LINE_INFO);
     }
 
+    struct LibCurlDetails
+    {
+        bool has_libcurl;
+        std::string version;
+    };
+
+    LibCurlDetails detect_libcurl()
+    {
+        // Determine if libcurl is installed in the system by attempting to load it
+        // At the moment we don't do anything with it, but we're tracking availability
+        // of libcurl to replace the current download/upload implementation
+#if defined(TEST_LIBCURL_AVAILABLE)
+        const auto libcurl_names = {
+            "libcurl.so",
+            "libcurl.so.4",
+            "libcurl.so.3",
+            "libcurl.so.2",
+            "libcurl.so.1",
+        };
+
+        for (const auto& libcurl_name : libcurl_names)
+        {
+            if (auto handle = dlopen(libcurl_name, RTLD_LAZY))
```

Comment:
It's not clear if we should `RTLD_LAZY` here given that the whole purpose here is to see if the copy of libcurl we get is usable, and if we're using it we're going to be calling a lot more than `curl_version`. In particular, we need curl to be a 'firewall' for the TLS provider (be that openssl or boringssl or whatever).